From: Rafael J. Wysocki <rjw@suse.de>
Subject: ACPI: Disable Windows 7 and Vista compatibility for Dell Vostro V13
Patch-mainline: No
References: bnc#602838

The declaration of Windows 7 and Vista compatibility via _OSI makes
Dell Vostro V13 misbehave if screen mode change is attempted by
pressing fn + F8, which works correcty with SLE11.  For this reason,
blacklist Dell Vostro V13 so that we don't claim Windows 7 or
Vista compatibility for this machine.

Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 drivers/acpi/blacklist.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

Index: linux-2.6.32-SLE11-SP1/drivers/acpi/blacklist.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/acpi/blacklist.c
+++ linux-2.6.32-SLE11-SP1/drivers/acpi/blacklist.c
@@ -191,6 +191,13 @@ static int __init dmi_disable_osi_win7(c
 	acpi_osi_setup("!Windows 2009");
 	return 0;
 }
+static int __init dmi_disable_osi_vista_and_win7(const struct dmi_system_id *d)
+{
+	printk(KERN_NOTICE PREFIX "DMI detected: %s\n", d->ident);
+	acpi_osi_setup("!Windows 2006");
+	acpi_osi_setup("!Windows 2009");
+	return 0;
+}
 
 static struct dmi_system_id acpi_osi_dmi_table[] __initdata = {
 	{
@@ -282,6 +289,14 @@ static struct dmi_system_id acpi_osi_dmi
 		     DMI_MATCH(DMI_PRODUCT_NAME, "Satellite P305D"),
 		},
 	},
+	{
+	.callback = dmi_disable_osi_vista_and_win7,
+	.ident = "Dell Vostro V13",
+	.matches = {
+		     DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
+		     DMI_MATCH(DMI_PRODUCT_NAME, "Vostro V13"),
+		},
+	},
 
 	/*
 	 * BIOS invocation of _OSI(Linux) is almost always a BIOS bug.
